### ISSUE TITLE
Members have all other members jobs

### ISSUE THREAD

**Issue by kinow** [BSC-ES/autosubmit#1147]
In GitLab by @rocsalvadorbsc on Oct 19, 2023, 14:47

Hello @dbeltrankyl and @kinow,

### Autosubmit Version

autosubmit/4.0.96-foss-2021b-Python-3.9.6

### Expid affected(If applicable)

a6ja

### Summary

When using `4.0.96` all the members have all the other's members jobs. This didn't happen in `4.0.88` where the memebers' jobs are created as defined in `/esarchive/autosubmit/a6ja/proj/auto-performance-profiling/autosubmit/configuration/platform/hpc2020/members`. 

### Steps to reproduce

Create the experiment with the `4.0.96` version and the`4.0.88`  and compare the plots.

### What is the current bug behavior?

The jobs created differ between versions.

### What is the expected correct behavior?

The jobs created should be as in `4.0.88`.

### Relevant logs and/or screenshots(if applicable)

- `4.0.96` plot:

`/esarchive/autosubmit/a6ja/plot/a6ja_20231019_0938.pdf`

- `4.0.88` plot:

`/esarchive/autosubmit/a6ja/plot/a6ja_20231019_0915.pdf`

**Comment by kinow** [comment]
In GitLab by @dbeltrankyl on Oct 19, 2023, 15:51

Hello @rocsalvadorbsc ,

Thanks for the report,

I'll revise it tomorrow as we're currently in a workshop. It is not blocking you as you can work with 88 for now, no?

**Comment by kinow** [comment]
In GitLab by @rocsalvadorbsc on Oct 19, 2023, 16:01

Yes, no worries, I can keep working.

**Comment by kinow** [comment]
In GitLab by @dbeltrankyl on Oct 25, 2023, 09:50

Hello @rocsalvadorbsc ,

This should be working in v4.0.98. The issue was that, in some places, you had an extra space between the list of members in the filter part, which was messing with the function. I've normalized the function, so you don't need a perfect matching syntax to work.

edit: The bug was introduced because we changed the way of checking that filter because it was giving issues with the chunks
